srcpkgs/ffplay/template

```diff
@@ -0,0 +1,47 @@
+# Template file for 'ffplay'
+# should be kept in sync with 'ffmpeg' (shared distfiles)
+pkgname=ffplay
+version=4.3.2
+wrksrc=ffmpeg-${version}
+revision=3
+hostmakedepends="pkg-config yasm"
+makedepends="ffmpeg-devel SDL2-devel"
+short_desc="Simple video player using FFmpeg and SDL2"
+maintainer="Orphaned <orphan@voidlinux.org>"
+license="GPL-3.0-or-later"
+homepage="https://www.ffmpeg.org"
+changelog="https://raw.githubusercontent.com/FFmpeg/FFmpeg/master/Changelog"
+distfiles="https://www.ffmpeg.org/releases/ffmpeg-${version}.tar.xz"
+checksum=46e4e64f1dd0233cbc0934b9f1c0da676008cad34725113fb7f802cfa84ccddb
+
+do_configure() {
+	# just to generate config.h
+	if [ "$CROSS_BUILD" ]; then
+		case "$XBPS_TARGET_MACHINE" in
+			arm*) _arch="arm";;
+			aarch64*) _arch="aarch64";;
+			mips*) _arch="mips";;
+			ppc64*) _arch="ppc64";;
+			ppc*) _arch="ppc";;
+			*) _arch="${XBPS_TARGET_MACHINE%%-musl}";;
+		esac
+
+		_args+=" --enable-cross-compile
+			--sysroot=$XBPS_CROSS_BASE
+			--cross-prefix=${XBPS_CROSS_TRIPLET}-
+			--target-os=linux --arch=${_arch}"
+	fi
+	./configure --prefix=/usr --disable-stripping --disable-everything ${_args}
+}
+
+do_build() {
+	set +x
+	$CC -o ffplay fftools/ffplay.c fftools/cmdutils.c \
+		-lm -I. \
+		$(pkg-config --cflags --libs sdl2 libavcodec libavdevice libavfilter libavformat libavresample libavutil libpostproc libswresample libswscale)
```

Comment:
This feels somewhat fragile, since we could miss any defines added by configure scripts and what not. I'd almost say building the entire thing and just taking the binary afterwards is safer.